src/compile-uri/expand-uri-template.js

```diff
@@ -0,0 +1,77 @@
+const ut = require('uri-template');
+
+module.exports = function expandURITemplate(uriTemplate, params) {
+  let parsed;
+  const result = {
+    errors: [],
+    warnings: [],
+    uri: null
+  };
+
+  try {
+    parsed = ut.parse(uriTemplate);
+  } catch (e) {
+    result.errors.push(`\
+Failed to parse URI template: ${uriTemplate}
+Error: ${e}\
+`);
```

Comment:
Hmm, ugly multi-line strings. We should use [`dedent`](https://github.com/dmnd/dedent), but feel free to leave this as a next step.